sunscreen_math/src/metal_impl/ristrettovec.rs

```diff
@@ -0,0 +1,716 @@
+use curve25519_dalek::{edwards::EdwardsPoint, ristretto::RistrettoPoint, CannonicalFieldElement};
+use metal::Buffer;
+
+use core::slice;
+use std::{
+    mem::size_of,
+    ops::{Add, Mul, Sub},
+};
+
+use crate::metal_impl::{ScalarVec, U32Arg};
+
+use super::Runtime;
+
+pub struct RistrettoPointVec {
+    data: Buffer,
+    len: usize,
+}
+
+impl RistrettoPointVec {
+    /// Creates a new [RistrettoVec].
+    ///
+    /// # Remarks
+    /// This code assumes the following layout of curve25519-dalek datastructures:
+    /// ```rust
+    /// struct RistrettoPoint(EdwardsPoint);
+    ///
+    /// struct EdwardsPoint {
+    ///     X: FieldElement2625,
+    ///     Y: FieldElement2625,
+    ///     Z: FieldElement2625,
+    ///     T: FieldElement2625,
+    /// }
+    ///
+    /// struct FieldElement2625([u32; 10]);
+    /// ```
+    /// To achieve this layout, you must use the u32 backend.
+    ///
+    pub fn new(x: &[RistrettoPoint]) -> Self {
+        let runtime = Runtime::get();
+
+        let len = x.len();
+
+        assert_eq!(size_of::<RistrettoPoint>(), size_of::<u32>() * 40);
+        let byte_len = x.len() * size_of::<RistrettoPoint>();
+        let data = runtime.alloc(byte_len);
+
+        let mut field_vec = Self { data, len };
+
+        let data_slice = field_vec.buffer_slice_mut();
+
+        for (i, p) in x.iter().enumerate() {
+            let x = p.0.X.to_u29();
+            let y = p.0.Y.to_u29();
+            let z = p.0.Z.to_u29();
+            let t = p.0.T.to_u29();
```

Comment:
I'm just curious: what is in the other fields of the `RistrettoPoint` (since `.0` was used)?

Reply:
None, it's a simple NewType. That's how you access the one field.

```
struct Foo(Bar)

// my_foo.0 returns the `Bar`
```